configure.ac

```diff
@@ -365,7 +365,11 @@ AS_IF([test x"$enable_unix_lib" = "xno" -o x"$enable_str_lib" = "xno"],
 # Allow the MSVC linker to be found even if ld isn't installed.
 # User-specified LD still takes precedence.
 AC_CHECK_TOOLS([LD],[ld link])
+# libtool expects host_os=mingw for native Windows
```

Comment:
Since the autoconf, I've allowed getting `--disable-shared` to work as expected to form part of this GPR. That has also expanded on a problem @shindere identified that `--enable-shared` doesn't work on MSVC (it does work for Cygwin and the mingw-w64 compilers). I did some in-depth poking into the code generated by `LT_INIT` and determined that the problem is that autoconf appears to assume that "native Windows" will be an OS called mingw - there is code in `LT_INIT` which will happily detect the output of `cl`, but it can't work with `$host_os` = `windows`.

This is making me wonder if it would be better for the MSVC ports to be identified not by `pc-windows` but by `pc-mingw32` or `cl-mingw32` or `msvc-mingw32` - i.e. use a different part of the triple to identify it? I don't know - and haven't yet made any effort to find out - what other projects have done about this, but it seems worth determining this now, before `i686-pc-windows` becomes the only way of identifying MSVC.

Thoughts @shindere?

Reply:
That sounds fishy. Is autoconf really that ill-adapted to MSVC toolchains? What is $host_os when we run on cygwin?

Reply:
It's not quite the right question - autoconf has reasonable support for `cl` as a compiler, but it does expect a Unix-like OS to be running on ... i.e. it expects the OS to be either `cygwin` or `mingw` (they're just the only two it's aware of). So it's objecting to (or rather, ignoring!) `windows` as the declared OS.